lib/tbd/psi.rb

```diff
@@ -1866,13 +1897,20 @@ def process(model = nil, argh = {})
         is[:corner          ] = set.keys.to_s.include?("corner")
         is[:parapet         ] = set.keys.to_s.include?("parapet")
         is[:roof            ] = set.keys.to_s.include?("roof")
+        is[:ceiling         ] = set.keys.to_s.include?("ceiling")
         is[:party           ] = set.keys.to_s.include?("party")
         is[:grade           ] = set.keys.to_s.include?("grade")
         is[:balcony         ] = set.keys.to_s.include?("balcony")
         is[:balconysill     ] = set.keys.to_s.include?("balconysill")
         is[:balconydoorsill ] = set.keys.to_s.include?("balconydoorsill")
         is[:rimjoist        ] = set.keys.to_s.include?("rimjoist")
 
+        if is.empty?
+          is[:head] = set.keys.to_s.include?("head")
+          is[:sill] = set.keys.to_s.include?("sill")
+          is[:jamb] = set.keys.to_s.include?("jamb")
+        end
```

Comment:
Nothing to do with plenum floors/ceilings. Simply wiser to post-process more generic tags.